src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,128 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private Node<K,V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (table.length * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
+        int hash = hash(key);
+        int bucket = hash % table.length;
+        Node<K,V> newNode = new Node<>(hash, key, value, null);
+        Node<K, V> checkNode;
+
+        // check collision
+        checkNode = table[bucket];
+
+        if (checkNode == null) {
+            table[bucket] = newNode;
+            size++;
+        } else {
+            do {
+                if (compareSolution(checkNode, key)) {
+                    checkNode.value = value;
+                    break;
+                }
 
+                if (checkNode.next == null) {
+                    checkNode.next = newNode;
+                    size++;
+                    break;
+                } else {
+                    checkNode = checkNode.next;
+                }
+            } while (true);
+        }
+        if (size >= threshold) {
+            resize();
+        }
```

Comment:
Better to resize before input. It's laways possible that it was last insert operation by user. Why we need to resize in such case